Branch-SDK/Branch-SDK/BranchUniversalObject.h

```diff
@@ -0,0 +1,47 @@
+//
+//  BranchUniversalObject.h
+//  Branch-TestBed
+//
+//  Created by Derrick Staten on 10/16/15.
+//  Copyright © 2015 Branch Metrics. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import "BranchLinkProperties.h"
+#import "Branch.h"
+
+@interface BranchUniversalObject : NSObject
+
+typedef NS_ENUM(NSInteger, ContentIndexMode) {
+    ContentIndexModePublic,
+    ContentIndexModePrivate
+};
+
+@property (nonatomic, strong) NSString *canonicalIdentifier;
+@property (nonatomic, strong) NSString *title;
+@property (nonatomic, strong) NSString *contentDescription;
+@property (nonatomic, strong) NSString *imageUrl;
+// Note: properties found in metadata will overwrite properties on the BranchUniversalObject itself
+@property (nonatomic, strong) NSDictionary *metadata;
+@property (nonatomic, strong) NSString *type;
+@property (nonatomic) ContentIndexMode contentIndexMode;
+@property (nonatomic, strong) NSArray *keywords;
+@property (nonatomic, strong) NSDate *expirationDate;
+
+- (instancetype)initWithCanonicalIdentifier:(NSString *)canonicalIdentifier;
+- (instancetype)initWithTitle:(NSString *)title;
+- (void)addMetadataKey:(NSString *)key value:(NSString *)value;
+- (void)registerView;
+- (void)registerViewWithCallback:(callbackWithParams)callback;
+- (NSString *)getShortUrlWithLinkProperties:(BranchLinkProperties *)linkProperties;
+- (void)getShortUrlWithLinkProperties:(BranchLinkProperties *)linkProperties andCallback:(callbackWithUrl)callback;
+- (UIActivityItemProvider *)getBranchActivityItemWithLinkProperties:(BranchLinkProperties *)linkProperties;
+- (void)showShareSheetWithShareText:(NSString *)shareText;
+- (void)showShareSheetWithLinkProperties:(BranchLinkProperties *)linkProperties andShareText:(NSString *)shareText fromViewController:(UIViewController *)viewController;
```

Comment:
Let's make the share sheet callback a requirement for both of these methods. Every time I've used it, I always put some completion code in the callback.